pkg/console/install_panels.go

```diff
@@ -1259,15 +1259,8 @@ func addInstallPanel(c *Console) error {
 				c.config.TTY = getFirstConsoleTTY()
 			}
 
-			requireGPT, err := isForceGPTRequired(c.config.Install.Device)
-			if err != nil {
-				logrus.Errorf("failed to determine forcing GPT is required: %s. Not forcing GPT.", err)
-				requireGPT = false
-			}
-			c.config.Install.ForceGPT = requireGPT
-			if c.config.Install.ForceGPT {
-				logrus.Infof("forcing GPT partition scheme")
-			}
+			// Custom partition table requires GPT
+			c.config.Install.ForceGPT = true
```

Comment:
@bk201 is there going to be compatible issues? If so we need to add it to the release note.

Reply:
According to [this](https://wiki.archlinux.org/title/Partitioning#Tricking_old_BIOS_into_booting_from_GPT), this would only affect some old BIOSes (Before 2010). We could still add it to the release note.

Reply:
will we also force the GPT partition of the PXE installation via this?

Reply:
Yes, as PXE installation begins from rendering this `addInstallPanel` panel ([REF](https://github.com/harvester/harvester-installer/blob/be59d22082c0a0de4789ffba8cf6113ebc7d1233/pkg/console/install_panels.go#L59-L62)). I also run a test and confirms this new partition layout will be applied for PXE installation.

Reply:
Dropping MBR support sounds bad. Can users opt-out of it when necessary?

Reply:
@gitlawr  Currently no. Adding additional Longhorn partition for MBR requires a huge amount of work because cOS does not support this.

If we do worry about forcing GPT, I guess the other way around is to opt out of creating an additional Longhorn partition.

Reply:
Probably drop MBR support is not ideal, so the best option for now is probably to `opt-out` the longhorn partition feature of the MBR boot, also we can file an issue on the cOS of the MRB support so we can leverage it in the future?

Reply:
@guangbochen  @gitlawr Please help check the following table as we would need to error out when forcing MBR in certain scenario:

<table>
    <tbody>
        <tr>
            <td></td>
            <td></td>
            <td><b>Forcing MBR</b></td>
            <td><b>Default</b></td>
        </tr>
        <tr>
            <td rowspan=2><b>UEFI</b></td>
            <td><b>Disk &gt MBR Limit</b></td>
            <td>ERR - Disk too large for MBR</td>
            <td>OK - Use GPT</td>
        </tr>
        <tr>
            <td><b>Disk &lt MBR Limit</b></td>
            <td>ERR - UEFI requires GPT</td>
            <td>OK - Use GPT</td>
        </tr>
        <tr>
            <td rowspan=2><b>BIOS</b></td>
            <td><b>Disk &gt MBR Limit</b>t</td>
            <td>ERR - Disk too large for MBR</td>
            <td>OK - Use GPT</td>
        </tr>
        <tr>
            <td><b>Disk &lt MBR Limit</b></td>
            <td>OK - Use MBR</td>
            <td>OK - Use GPT</td>
        </tr>
    </tbody>
</table>

Reply:
I'm just wondering if it is better to allow users to config it as an option instead of force it automatically, e.g even if they boot from the BIOS with a disk size less than 2TB they can still use GPT mode if it was supported:
- always default to the `GPT` partition of both mode and UEFI will be forced to GPT only(e.g if `/sys/firmware/efi` exist)
- add a config option to specify the partition mode, i.e MBR or GPT only
- add a grub menu to boot with MBR partition(not default)

Reply:
Will add an option into Installer UI for users to opt-out forcing GPT.